frontend/public/components/row-filters.tsx

```diff
@@ -0,0 +1,348 @@
+  const [filterType, setFilterType] = React.useState(FilterType.NAME);
+  const [isOpen, setIsOpen] = React.useState(false);
+  const [selected, setSelected] = React.useState(() => getSelectedRowFilters(rowFilters));
+
+  const filters: Filter = React.useMemo(
+    () =>
+      rowFilters.reduce((acc, curr) => {
+        const items = _.map(curr.items, 'id');
+        acc[curr.filterGroupName] = items;
+        return acc;
+      }, {}),
+    [rowFilters],
+  );
+
+  // { 'Xstatus' = 'x-y-status' ...}
+  const filterKeys: FilterKeys = React.useMemo(
+    () =>
+      rowFilters.reduce((acc, curr) => {
+        const str = `${storagePrefix}${curr.type}`;
+        acc[curr.filterGroupName] = str;
+        return acc;
+      }, {}),
+    [rowFilters],
+  );
+
+  // Parse state from URL
+  React.useEffect(() => {
+    let q: string = '',
+      name: string;
+    const rowFiltersFromURL: string[] = [];
+
+    if (window.location.search) {
+      const params = new URLSearchParams(window.location.search);
+      q = params.get('q');
+      name = params.get('name');
+      _.map(_.values(filterKeys), (f) => {
+        const vals = params.get(f);
+        if (vals) {
+          rowFiltersFromURL.push(...vals.split(','));
+        }
+      });
+    }
+    const labels = q ? q.split(',') : [];
+    labels.length > 0 && setLabelFilters(labels);
+    setNameFilter(name || '');
+    setSelected([...selected, ...rowFiltersFromURL]);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  /* Logic for Name and Label Filter */
+
+  const applyFilter = (value: string | string[], type: FilterType) => {
+    const filter = type === FilterType.NAME ? value : { all: value };
+    props.reduxIDs.forEach((id) => props.filterList(id, filterTypeMap[type], filter));
+  };
+
+  const addLabelFilter = (filter: string) => {
+    setLabelFilters([...labelFilters, filter]);
+    setQueryArgument('q', [...labelFilters, filter].join(','));
+    applyFilter([...labelFilters, filter], FilterType.LABEL);
+    setLabelFilterText('');
+  };
+
+  const removeLabelFilter = (filter: string) => {
+    const filtered = labelFilters.filter((l) => l !== filter);
+    setLabelFilters(filtered);
+    setQueryArgument('q', filtered.join(','));
+    if (filtered.length === 0) {
+      removeQueryArgument('q');
+    }
+    applyFilter(filtered || [], FilterType.LABEL);
+    setLabelFilterText('');
+  };
+
+  const removeAllLabelFilters = () => {
+    setLabelFilters([]);
+    setQueryArgument('q', '');
+    removeQueryArgument('q');
+    applyFilter([], FilterType.LABEL);
+    setLabelFilterText('');
+  };
+
+  const updateFilter = (filter: string, type: FilterType) => {
+    if (type === FilterType.NAME) {
+      setNameFilter(filter);
+      filter ? setQueryArgument('name', filter) : removeQueryArgument('name');
+      applyFilter(filter, FilterType.NAME);
+    } else {
+      addLabelFilter(filter);
+    }
+  };
+
+  const updateSearchFilter = (value: string) => {
```

Comment:
Can you clarify the different between `updateFilter` and `updateSearchFilter`?

Reply:
Found redundant logic. Terminated `updateFilter`